AppKit/CoreAnimation/CPAnimationContext.j

```diff
@@ -152,24 +152,29 @@ CPLog.debug(_cmd + "context stack =" + _CPAnimationContextStack);
 
     duration = [animation duration] || [self duration];
 
-    var needsFrameTimer = (aKeyPath == @"frame" || aKeyPath == @"frameSize") &&
-                          ([anObject hasCustomLayoutSubviews] || [anObject hasCustomDrawRect]) &&
-                          (objectId = [anObject UID]);
+    var needsPeriodicFrameUpdates = (aKeyPath === @"frame" || aKeyPath === @"frameSize" || aKeyPath === @"frameOrigin") &&
+                                    [[anObject animator] wantPeriodicFrameUpdates] &&
```

Comment:
The result here seems wrong if we have a custom layout and a frameOrigin keyPath. I'm afraid you will have to compute needsPeriodicFrameUpdates with the custom layout|drawing info here.
It's (isFrameOrFrameSizeKeyPath && isCustomLayoutOrDrawing) || wantPeriodicFrameUpdates.